dashboard/client/src/pages/serve/ServeDeploymentsListPage.tsx

```diff
@@ -191,16 +144,16 @@ export const ServeDeploymentsListPage = () => {
                   </TableRow>
                 </TableHead>
                 <TableBody>
-                  {filteredServeDeployments
+                  {serveApplications
                     .slice(
                       (page.pageNo - 1) * page.pageSize,
                       page.pageNo * page.pageSize,
                     )
-                    .map((deployment) => (
-                      <ServeDeploymentRow
-                        key={`${deployment.application.name}-${deployment.name}`}
-                        deployment={deployment}
-                        application={deployment.application}
+                    .map((application) => (
+                      <ServeApplicationRows
+                        key={`${application.name}__${application.deployments.length}`}
```

Comment:
Not confident about this `key`. @alanwguo what would you recommend? Does it make sense to use `deployments.length` here to cause some kind of refresh if the number of deployments changes?

Reply:
Thanks, fixed be06ee83e8